selfservice/strategy/code/strategy_recovery.go

```diff
@@ -0,0 +1,605 @@
+	//
+	// required: true
+	// format: uri
+	RecoveryLink string `json:"recovery_link"`
+
+	// RecoveryCode is the code that can be used to recover the account
+	//
+	// required: true
+	RecoveryCode string `json:"recovery_code"`
+
+	// Expires At is the timestamp of when the recovery flow expires
+	//
+	// The timestamp when the recovery link expires.
+	ExpiresAt time.Time `json:"expires_at"`
+}
+
+// swagger:route POST /admin/recovery/code v0alpha2 adminCreateSelfServiceRecoveryCode
+//
+// # Create a Recovery Link
+//
+// This endpoint creates a recovery code which should be given to the user in order for them to recover
+// (or activate) their account.
+//
+//	Consumes:
+//	- application/json
+//
+//	Produces:
+//	- application/json
+//
+//	Schemes: http, https
+//
+//	Responses:
+//		200: selfServiceRecoveryCode
+//		400: jsonError
+//		404: jsonError
+//		500: jsonError
+func (s *Strategy) createRecoveryCode(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	var p adminCreateSelfServiceRecoveryCodeBody
+	if err := s.dx.Decode(r, &p, decoderx.HTTPJSONDecoder()); err != nil {
+		s.deps.Writer().WriteError(w, r, err)
+		return
+	}
+
+	ctx := r.Context()
+	config := s.deps.Config()
+
+	expiresIn := config.SelfServiceCodeMethodLifespan(ctx)
+	if len(p.ExpiresIn) > 0 {
+		// If an expiration of the code was supplied use it instead of the default duration
+		var err error
+		expiresIn, err = time.ParseDuration(p.ExpiresIn)
+		if err != nil {
+			s.deps.Writer().WriteError(w, r, errors.WithStack(herodot.
+				ErrBadRequest.
+				WithReasonf(`Unable to parse "expires_in" whose format should match "[0-9]+(ns|us|ms|s|m|h)" but did not: %s`, p.ExpiresIn)))
+			return
+		}
+	}
+
+	if time.Now().Add(expiresIn).Before(time.Now()) {
+		s.deps.Writer().WriteError(w, r, errors.WithStack(herodot.ErrBadRequest.WithReasonf(`Value from "expires_in" must result to a future time: %s`, p.ExpiresIn)))
+		return
+	}
+
+	flow, err := recovery.NewFlow(config, expiresIn, s.deps.GenerateCSRFToken(r), r, s, flow.TypeBrowser)
+	if err != nil {
+		s.deps.Writer().WriteError(w, r, err)
+		return
+	}
+	flow.DangerousSkipCSRFCheck = true
+	flow.State = recovery.StateEmailSent
+	flow.UI.Nodes.Append(node.NewInputField("code", nil, node.CodeGroup, node.InputAttributeTypeNumber, node.WithRequiredInputAttribute).
+		WithMetaLabel(text.NewInfoNodeLabelVerifyOTP()),
+	)
+
+	flow.UI.Nodes.
+		Append(node.NewInputField("method", s.RecoveryStrategyID(), node.CodeGroup, node.InputAttributeTypeSubmit).
+			WithMetaLabel(text.NewInfoNodeLabelSubmit()))
+
+	if err := s.deps.RecoveryFlowPersister().CreateRecoveryFlow(ctx, flow); err != nil {
+		s.deps.Writer().WriteError(w, r, err)
+		return
+	}
+
+	id, err := s.deps.IdentityPool().GetIdentity(ctx, p.IdentityID)
+	if errors.Is(err, sqlcon.ErrNoRows) {
+		s.deps.Writer().WriteError(w, r, errors.WithStack(herodot.ErrBadRequest.WithReasonf("The requested identity id does not exist.").WithWrap(err)))
```

Comment:
I guess returning 404 would be more meaningful, as the request is not malformed?
```suggestion
		s.deps.Writer().WriteError(w, r, err)
```
You can then add more details further down in the persister.

Reply:
> You can then add more details further down in the persister.

What do you mean by that? 
I'd just change `ErrBadRequest` to `ErrNotFound`.

Reply:
`sqlcon.ErrNoRows` already is `herodot.ErrNotFound`. You can add more details by specifying that the identity was not found, instead of just the "generic" 404. This handler is about the code, but it needs the identity as a prerequisite. Adding something like `.WithReadson("requested identity does not exist")` makes it clear that the error is not related to the code itself.

Reply:
I guess here, as `GetIdentity` is used in many places. But whatever makes more sense. You can probably use `errors.As` to unwrap the herodot error.

Reply:
Done.